packages/react/src/transactions/adapters/responders/UpdateProfileImageResponder.ts

```diff
@@ -0,0 +1,67 @@
+import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
+import {
+  GetProfileDocument,
+  GetProfileQuery,
+  GetProfileQueryVariables,
+  MediaSetFragment,
+} from '@lens-protocol/api-bindings';
+import {
+  UpdateOffChainProfileImageRequest,
+  UpdateProfileImageRequest,
+} from '@lens-protocol/domain/use-cases/profile';
+import {
+  BroadcastedTransactionData,
+  ITransactionResponder,
+} from '@lens-protocol/domain/use-cases/transactions';
+
+function isUpdateOffChainProfileImageRequest(
+  request: UpdateProfileImageRequest,
+): request is UpdateOffChainProfileImageRequest {
+  return 'url' in request;
+}
+
+export class UpdateProfileImageResponder
+  implements ITransactionResponder<UpdateProfileImageRequest>
+{
+  constructor(private readonly client: ApolloClient<NormalizedCacheObject>) {}
+
+  async prepare({ request }: BroadcastedTransactionData<UpdateProfileImageRequest>) {
+    if (isUpdateOffChainProfileImageRequest(request)) {
+      const profileIdentifier = this.client.cache.identify({
+        __typename: 'Profile',
+        id: request.profileId,
+      });
+      this.client.cache.modify({
+        id: profileIdentifier,
+        fields: {
+          picture(oldPicture: MediaSetFragment) {
+            return {
+              ...oldPicture,
+              original: {
+                ...oldPicture.original,
+                url: request.url,
+              },
+            };
+          },
+        },
+      });
+    }
+  }
+
+  async commit(_: BroadcastedTransactionData<UpdateProfileImageRequest>) {
+    //
+  }
+
+  async rollback({ request }: BroadcastedTransactionData<UpdateProfileImageRequest>) {
+    // we don't know the previous profile image url, so just query the profile again
+    if (isUpdateOffChainProfileImageRequest(request)) {
+      await this.client.query<GetProfileQuery, GetProfileQueryVariables>({
+        query: GetProfileDocument,
+        variables: {
+          request: { profileId: request.profileId },
```

Comment:
Should we include the `observerId` so relevant edges are tailored to the current profile?

Thinking out loud: is it safe to use `request.profileId` or should we access the active profile id?

Reply:
I could also use the `Map<ProfileId, Url>` approach and just revert the URL in the cache, which one do you think it's better? I guess cache-only has a lower impact.

Reply:
in the end, it was way easier to implement a fresh query on commit and rollback than to store the URL value in the Map. Will that be enough?

Reply:
I think it's fine. I was thinking about scenarios where there are several updates related to a profile in the queue (e.g. display name, cover image, profile image)... but couldn't think of scenario that yields the sequencing of the `prepare/commit|rollback` in the wrong order.

In this comment I was asking a different thing though. I was entertaining the idea if we should include `observerId` in the query request var.

Reply:
oh yes, I added `observerId` too to the query (I just haven't pushed that yet). 
One has to be an activeUser to update their profile image so one will be also the observer. Could adding the observer harm any other use case?

Reply:
pushed now https://github.com/lens-protocol/lens-sdk/pull/72/commits/2037406cfc565783a868e4890a7dfa0b05c55701

Reply:
I think there is a chance that a user makes an update to the profile picture of one of the profiles they own... then switch the current active profile (T-1368 I just started) before the tx settles. By the time the commit/rollback happens the observer will be technically wrong.

The thing is that nobody might notice cause it affects fields that one is unlikely to care about when looking at their own profiles (e.g. `isFollowing(profileId: $observerId)`).

Let's not worry about this for now.